client/question/services/ConversationManagerServiceSpec.js

```diff
@@ -43,29 +39,61 @@ describe('ConversationManagerService', function() {
       python: []
     },
     testSuites: [{
-      id: 'GENERAL_CASE',
-      humanReadableName: 'the general case',
+      id: 'TASK1SUITE1',
+      humanReadableName: 'suite 1 -- single words',
       testCases: [{
-        input: 'task_1_correctness_test_1',
-        allowedOutputs: [true]
+        input: 'task1suite1test1',
+        allowedOutputs: ['task1suite1test1']
       }, {
-        input: 'task_1_correctness_test_2',
-        allowedOutputs: [true]
+        input: 'task1suite1test2',
+        allowedOutputs: ['task1suite1test2']
+      }]
+    }, {
+      id: 'TASK1SUITE2',
+      humanReadableName: 'suite 2 -- multiple words',
+      testCases: [{
+        input: 'task1 suite2 test1',
+        allowedOutputs: ['task1 suite2 test1']
+      }, {
+        input: 'task1 suite2 test2',
+        allowedOutputs: ['task1 suite2 test2']
       }]
     }],
     buggyOutputTests: [{
-      buggyFunctionName: 'AuxiliaryCode.mockAuxiliaryCodeOne',
+      buggyFunctionName: 'AuxiliaryCode.returnIAmBuggy',
       ignoredTestSuiteIds: [],
       messages: [
-        "Mock BuggyOutputTest Message One for task1",
-        "Mock BuggyOutputTest Message Two for task1",
-        "Mock BuggyOutputTest Message Three for task1"
+        'Mock BuggyOutputTest Message One for task1',
+        'Mock BuggyOutputTest Message Two for task1',
+        'Mock BuggyOutputTest Message Three for task1'
+      ]
+    }, {
+      buggyFunctionName: 'AuxiliaryCode.returnYesIfInputHasNoSpaces',
+      ignoredTestSuiteIds: [],
+      messages: [
+        'This buggy message should NOT trigger if code just returns Yes'
+      ]
+    }, {
+      buggyFunctionName: 'AuxiliaryCode.returnYesIfInputHasNoSpaces',
+      // TODO(sll): Remove GENERAL_CASE from here. This is due to an error in
```

Comment:
@eyurko, I want to draw your attention to this. When refactoring the tests, I discovered an error whereby suites in other tasks would also be considered when determining whether a test hits the buggy-output criterion. This is incorrect, because only suites _within a particular task_ should be considered when determining whether or not a given buggy output for the task is relevant.

We have not run into this issue before, because many of our questions, including the upcoming ones we're working on, are single-task. Fixing this issue is a fair amount of work because it requires an overhaul of the preprocessed code for detecting buggy output.

I am inclined to leave this as-is for the time being, and file an issue for it (in fact I'll just go ahead and do that anyway), since it is not an issue that will come up for a while (it only affects multi-task questions, and we haven't fully worked through the feedback flow for that in any case). However, I want to make sure you see this and are OK with it. If not, I'm happy to work on it and fix that in a separate PR before submitting this one.

Reply:
Filed #633.

Reply:
That's the right play, yeah.

Reply:
Just make sure we don't allow any multi-task questions before we fix this.

Reply:
Yep. I don't think we will do that before we sort out the feedback flow for multi-task questions anyway, at which point I'll need to do a similar audit to what I'm doing now. But this issue should be fixed well before then -- I know how to do it, it just takes a bit of time.

Thanks!